client/src/main/java/org/eclipse/hono/client/DisconnectListener.java

```diff
@@ -0,0 +1,30 @@
+/**
+ * Copyright (c) 2019 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+
+package org.eclipse.hono.client;
+
+
+/**
+ * A listener to be notified when a connection is lost unexpectedly.
+ *
+ */
+public interface DisconnectListener {
```

Comment:
I would annotate those with `@FunctionalInterface` … which lets the compiler enforce all requirements:

```suggestion
@FunctionalInterface
public interface DisconnectListener {
```